org.abapgit.adt.backend/src/org/abapgit/adt/backend/IRepositoryService.java

```diff
@@ -104,4 +102,64 @@ void repositoryChecks(IProgressMonitor monitor, IExternalRepositoryInfoRequest c
 	 */
 	URI getURIFromAtomLink(IRepository repository, String relation);
 
+	/**
+	 * Returns the locally modified objects for the given repository
+	 *
+	 * @param currRepository
+	 *            Repository for which the modified Objects are to be fetched
+	 * @param user
+	 *            User
+	 * @param password
+	 *            Repository Password
+	 * @param monitor
+	 *            Progress monitor
+	 * @return Modified Objects for the given repository
+	 */
+	IAbapGitPullModifiedObjects getModifiedObjects(IProgressMonitor monitor, IRepository currRepository,
+			String user, String password);
+
+	//Valid for older backend before 2105
+	/**
+	 * Performs pull action for the given repository and replace all the
+	 * modified objects
+	 *
+	 * @param existingRepository
+	 *            Repository for which the pull action is to be performed
+	 * @param branch
+	 *            Branch to be pulled
+	 * @param monitor
+	 *            Progress monitor
+	 * @param transportRequest
+	 *            Transport Request for the pulled objects
+	 * @param user
+	 *            User name
+	 * @param password
+	 *            Password for Repository
+	 */
+	IAbapObjects pullRepository(IRepository existingRepository, String branch, String transportRequest, String user, String password,
+			IProgressMonitor monitor);
+
+	//Valid for backend versions post 2105
+	/**
+	 * Performs pull action for the given repository and replace the selected
+	 * modified objects
+	 *
+	 * @param existingRepository
+	 *            Repository for which the pull action is to be performed
+	 * @param branch
+	 *            Branch to be pulled
+	 * @param monitor
+	 *            Progress monitor
+	 * @param transportRequest
+	 *            Transport Request for the pulled objects
+	 * @param user
+	 *            User name
+	 * @param password
+	 *            Password
+	 * @param selectedObjectsToPull
+	 *            Objects selected by the user to be pulled
+	 */
+	IAbapObjects pullRepository(IRepository existingRepository, String branch, String transportRequest, String user, String password,
```

Comment:
The pullRepository method is overridden here to handle backend call to pull selected objects,